cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -490,6 +573,68 @@ private void closeScanner() {
     }
   }
 
+  /**
+   * Filters elements matching an index  {@link ScannerIterator}.
+   */
+  @VisibleForTesting
+  static final class FilterByIndexIterator extends AbstractCloseableIterator<StructuredRow> {
+    private final CloseableIterator<StructuredRow> scannerIterator;
+    private final Field<?> filterIndex;
```

Comment:
filterIndex and schema are only used in the constructor, can remove these variables